aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -111,6 +130,91 @@ fn native_get_script_hash(
     )])
 }
 
+fn native_sender_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SENDER_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        Ok(smallvec![Value::address(transaction_context.sender())])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_secondary_signers_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SECONDARY_SIGNERS_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        let secondary_signers = transaction_context.secondary_signers();
+        context.charge(
+            TRANSACTION_CONTEXT_SECONDARY_SIGNERS_PER_SIGNER
+                * NumArgs::new(secondary_signers.len() as u64),
+        )?;
+        Ok(smallvec![Value::vector_address(secondary_signers)])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_fee_payer_opt_internal(
```

Comment:
Rename to gas_payer

Reply:
I would prefer to update all `fee_payer` occurrences in our codebase in this case, because it is confusing when we have `gas_payer` and `fee_payer` refer to the same thing? Or are they different?

Reply:
Renamed it to `gas_payer`